SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp

```diff
@@ -467,7 +465,44 @@ void UGlobalStateManager::QueryGSM(const QueryDelegate& Callback)
 	Receiver->AddEntityQueryDelegate(RequestID, GSMQueryDelegate);
 }
 
-void UGlobalStateManager::ApplyVirtualWorkerMappingFromQueryResponse(const Worker_EntityQueryResponseOp& Op)
+void UGlobalStateManager::QueryTranslation()
+{
+	if (bTranslationQueryInFlight)
+	{
+		// Only allow one in flight query. Retries will be handled by the SpatialNetDriver.
+		return;
+	}
+
+	// Build a constraint for the Virtual Worker Translation.
+	Worker_ComponentConstraint TranslationComponentConstraint{};
+	TranslationComponentConstraint.component_id = SpatialConstants::VIRTUAL_WORKER_TRANSLATION_COMPONENT_ID;
+
+	Worker_Constraint TranslationConstraint{};
+	TranslationConstraint.constraint_type = WORKER_CONSTRAINT_TYPE_COMPONENT;
+	TranslationConstraint.constraint.component_constraint = TranslationComponentConstraint;
+
+	Worker_EntityQuery TranslationQuery{};
+	TranslationQuery.constraint = TranslationConstraint;
+	TranslationQuery.result_type = WORKER_RESULT_TYPE_SNAPSHOT;
+
+	Worker_RequestId RequestID = NetDriver->Connection->SendEntityQueryRequest(&TranslationQuery);
+
+	EntityQueryDelegate TranslationQueryDelegate;
+	TranslationQueryDelegate.BindLambda([this](const Worker_EntityQueryResponseOp& Op)
```

Comment:
Perhaps worth binding a weakobjptr here instead of this, in case the gsm is destroyed

Reply:
Done. We do the bind to this a few other places in the code, and I only fixed this one for now and I created https://improbableio.atlassian.net/browse/UNR-3743 to make a pass over the code and clean up others.